File: FitnessManagementSystem/Front_End/src/pages/Register/TrainerRegister.js
import { useEffect, useState } from "react";
import i1 from "./11.jpg";
import i2 from "./12.webp";
import i3 from "./16.jpg";
import i4 from "./14.webp";
import i5 from "./15.jpg";
import i6 from "./13.jpg";
import axios from "axios";

function TrainerRegister() {
  const [fullname, setFullname] = useState("");
  const [speciality, setSpeciality] = useState("");
  const [email, setEmail] = useState("");
  const [contact, setContact] = useState("");
  const [username, setusername] = useState("");
  const [password, setPassword] = useState("");
  const [fees, setFees] = useState("");
  const [status, setStatus] = useState("");
  const [workexp, setWorkexp] = useState("");
  const [messege, setMessege] = useState("");

  function addData() {
    const user = {};
    user.fullname = fullname;
    user.workexp = workexp;
    user.speciality = speciality;
    user.fees = fees;
    user.email = email;
    user.contactno = contact;
    user.username = username;
    user.password = password;
    user.status = status;

    console.log(user);

    if (
      fullname == "" ||
      username == "" ||
      password == "" ||
      status == "" ||
      email == "" ||
      contact == "" ||
      workexp == "" ||
      speciality == "" ||
      fees == ""
    ) {
      setMessege("Fill Necessary Details");
      return;
    }

    axios
      .post("http://localhost:8080/registertrainer", user)
      .then((response) => {
        setMessege(response.data);
      })
      .catch((error) => {
        setMessege(error.data);
      });
  }

  return (
    <section>
      <div className="container py-5 h-100">
        <div className="row d-flex justify-content-center align-items-center h-100">
          <div className="col col-xl-10">
            <div className="card" style={{ borderRadius: " 1rem" }}>
              <div className="row g-0">
                <div className="col-md-6 col-lg-5 d-none d-md-block">
                  <img
                    src={i1}
                    alt="login form"
                    className="img-fluid"
                    style={{ borderRadius: " 1rem 0 0 1rem" }}
                  />
                  <br />
                  <br />
                  <br />
                  <img
                    src={i2}
                    alt="login form"
                    className="img-fluid"
                    style={{ borderRadius: " 1rem 0 0 1rem" }}
                  />
                  <br />
                  <br />
                  <br />
                  <img
                    src={i3}
                    alt="login form"
                    className="img-fluid"
                    style={{ borderRadius: " 1rem 0 0 1rem" }}
                  />
                  <br />
                  <br />
                  <br />
                  <img
                    src={i4}
                    alt="login form"
                    className="img-fluid"
                    style={{ borderRadius: " 1rem 0 0 1rem" }}
                  />
                  <br />
                  <br />
                  <br />
                  <img
                    src={i6}
                    alt="login form"
                    className="img-fluid"
                    style={{ borderRadius: " 1rem 0 0 1rem" }}
                  />
                  <br />
                  <br />
                  <br />
                  <img
                    src={i5}
                    alt="login form"
                    className="img-fluid"
                    style={{ borderRadius: " 1rem 0 0 1rem" }}
                  />
                </div>
                <div className="col-md-6 col-lg-7 d-flex align-items-center">
                  <div
                    className="card-body p-4 p-lg-5 text-black"
                    style={{ backgroundColor: "rgb(5, 101, 134)" }}
                  >
                    <form>
                      <a
                        className="form-select btn btn-dark btn-lg btn-block"
                        href="/memberregistration"
                      >
                        Member Registration
                      </a>
                      {/* <a className="form-select btn btn-dark btn-lg btn-block" href="/adminregistration" >Admin Registration</a> */}
                      <div className="d-flex align-items-center mb-3 pb-1">
                        <i
                          className="fas fa-key fa-2x me-3"
                          style={{ color: " #000" }}
                        ></i>
                        <span className="h1 fw-bold mb-0">Tainer Register</span>
                      </div>
                      <div className="form-outline mb-4">
                        <label className="form-label">Username</label>
                        <input
                          type="text"
                          placeholder="enter your username"
                          className="form-control form-control-lg"
                          required
                          minLength={8}
                          maxLength={16}
                          onBlur={(e) => {
                            setusername(e.target.value);
                          }}
                        />
                      </div>
                      <div className="form-outline mb-4">
                        <label className="form-label">Password</label>
                        <input
                          type="password"
                          placeholder="enter your password"
                          className="form-control form-control-lg"
                          required
                          minLength={8}
                          maxLength={16}
                          onBlur={(e) => {
                            setPassword(e.target.value);
                          }}
                        />
                      </div>
                      <div className="form-outline mb-4">
                        <label className="form-label">Full Name</label>
                        <input
                          type="text"
                          placeholder="enter your full name"
                          className="form-control form-control-lg"
                          required
                          minLength={1}
                          maxLength={16}
                          onBlur={(e) => {
                            setFullname(e.target.value);
                          }}
                        />
                      </div>
                      <div className="form-outline mb-4">
                        <label className="form-label">Speciality</label>
                        <input
                          type="text"
                          placeholder="enter your speciality"
                          className="form-control form-control-lg"
                          required
                          minLength={1}
                          maxLength={16}
                          onBlur={(e) => {
                            setSpeciality(e.target.value);
                          }}
                        />
                      </div>
                      <div className="form-outline mb-4">
                        <label className="form-label">Work Experience</label>
                        <input
                          type="number"
                          placeholder="enter your work experience"
                          className="form-control form-control-lg"
                          required
                          minLength={1}
                          maxLength={16}
                          onBlur={(e) => {
                            setWorkexp(e.target.value);
                          }}
                        />
                      </div>
                      <div className="form-outline mb-4">
                        <label className="form-label">Fees</label>
                        <input
                          type="number"
                          placeholder="enter your fees"
                          className="form-control form-control-lg"
                          required
                          minLength={1}
                          maxLength={16}
                          onBlur={(e) => {
                            setFees(e.target.value);
                          }}
                        />
                      </div>
                      <div className="form-outline mb-4">
                        <label className="form-label">E-mail</label>
                        <input
                          type="mail"
                          placeholder="enter your email"
                          className="form-control form-control-lg"
                          required
                          minLength={5}
                          maxLength={70}
                          onBlur={(e) => {
                            setEmail(e.target.value);
                          }}
                        />
                      </div>
                      <div className="form-outline mb-4">
                        <label className="form-label">Contact</label>
                        <input
                          type="number"
                          placeholder="enter your contact no"
                          className="form-control form-control-lg"
                          required
                          minLength={3}
                          maxLength={16}
                          onBlur={(e) => {
                            setContact(e.target.value);
                          }}
                        />
                      </div>
                      <div>
                        <label className="form-label">Status</label>
                        <select
                          className="form-select btn btn-dark btn-lg btn-block"
                          aria-label="Default select example"
                          onChange={(e) => {
                            setStatus(e.target.value);
                          }}
                        >
                          <option>Select</option>
                          <option value="active">Active</option>
                          <option value="inactive">InActive</option>
                        </select>
                      </div>
                      <b style={{ color: "red" }}>{messege}</b>
                      <div className="pt-1 mb-4">
                        <button
                          className="btn btn-dark btn-lg btn-block"
                          type="button"
                          onClick={addData}
                        >
                          Register
                        </button>
                      </div>
                      <div className="pt-1 mb-4">
                        <button
                          className="btn btn-dark btn-lg btn-block"
                          type="Reset"
                        >
                          Reset
                        </button>
                      </div>
                      <a style={{ color: "black" }} href="/forgetpassword">
                        Forgot password?
                      </a>
                      <p className="mb-5 pb-lg-2" style={{ color: " #fff" }}>
                        Already't have an account?{" "}
                        <a href="/login" style={{ color: " #000" }}>
                          Login here
                        </a>
                      </p>
                      <a href="/termsAndCondition" style={{ color: "black" }}>
                        Terms and Conditions.
                      </a>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  );
}

export default TrainerRegister;

File: FitnessManagementSystem/Back_End/src/main/java/Fitness/FitnessProject/models/TrainerRegister.java
package Fitness.FitnessProject.models;

public class TrainerRegister {
    int tid;
	String username;
	String password;
	String fullname;
	Double fees;
	String email;
	String speciality;
	int workexp;
	String status;
	int contactno;
	String role;
	int loginid;

	public TrainerRegister(int tid,int contactno, String email, Double fees, String fullname, String speciality,String status,  int workexp,
			 int loginid) {
		super();
		this.tid = tid;
		this.fullname = fullname;
		this.fees = fees;
		this.email = email;
		this.speciality = speciality;
		this.workexp = workexp;
		this.status = status;
		this.contactno = contactno;
		this.loginid = loginid;
	}

	public String getRole() {
		return role;
	}

	public void setRole(String role) {
		this.role = role;
	}

	public int getLoginid() {
		return loginid;
	}

	public void setLoginid(int loginid) {
		this.loginid = loginid;
	}

	public TrainerRegister() {
		super();
	}
	
	public TrainerRegister(int tid,int contactno,  String email, Double fees,String fullname, String speciality, 
			String status, int workexp, int loginid,String password,String role,String username) {
		super();
		this.tid = tid;
		this.fullname = fullname;
		this.fees = fees;
		this.email = email;
		this.speciality = speciality;
		this.workexp = workexp;
		this.status = status;
		this.contactno = contactno;
		this.loginid = loginid;
		this.password= password;
		this.role=role;
		this.username=username;
	}



	public TrainerRegister(String username, String password, String fullname, Double fees, String email,
			String speciality, int workexp, String status, int contactno) {
		super();
		this.username = username;
		this.password = password;
		this.fullname = fullname;
		this.fees = fees;
		this.email = email;
		this.speciality = speciality;
		this.workexp = workexp;
		this.status = status;
		this.contactno = contactno;
	}
	
	public int getTid() {
		return tid;
	}
	public void setTid(int tid) {
		this.tid = tid;
	}
	public String getUsername() {
		return username;
	}
	public void setUsername(String username) {
		this.username = username;
	}
	public String getPassword() {
		return password;
	}
	public void setPassword(String password) {
		this.password = password;
	}
	public String getFullname() {
		return fullname;
	}
	public void setFullname(String fullname) {
		this.fullname = fullname;
	}
	public Double getFees() {
		return fees;
	}
	public void setFees(Double fees) {
		this.fees = fees;
	}
	public String getEmail() {
		return email;
	}
	public void setEmail(String email) {
		this.email = email;
	}
	public String getSpeciality() {
		return speciality;
	}
	public void setSpeciality(String speciality) {
		this.speciality = speciality;
	}
	public int getWorkexp() {
		return workexp;
	}
	public void setWorkexp(int workexp) {
		this.workexp = workexp;
	}
	public String getStatus() {
		return status;
	}
	public void setStatus(String status) {
		this.status = status;
	}
	public int getContactno() {
		return contactno;
	}
	public void setContactno(int contactno) {
		this.contactno = contactno;
	}
	
	
}
